### ISSUE TITLE
Counter counts wrong (total failed)

### ISSUE THREAD

**Issue by pisoto** [fail2ban/fail2ban#2506]
Hi, I tried to set a fail2ban with ssh, but every time I entered the wrong password, two fails were added to the total failed, instead of only one. Every two fails, I get banned.  
I don't understand why,  because I configured to get a ban after three fails and I get banned after two fails.

`sudo fail2ban-client status sshd`
`Status for the jail: sshd`
`|- Filter`
`|  |- Currently failed:	0`
`|  |- Total failed:	2`
`|  - File list:	/var/log/auth.log`
`- Actions`
 `  |- Currently banned:	1`
`   |- Total banned:	1`
   `- Banned IP list:	192.168.43.250`

the jail.local file config
`[sshd]`
`enabled = true`
`filter = sshd`
`port = 22`
`logpath = /var/log/auth.log`
`maxretry = 3`
 
to sum up, I want to block the host after three wrong passwords, and after the ban ends, I want to let the user try three times again.

p.s
Why does the "currently failed" not increase? And only the total failed increases?

**Comment by sebres** [comment]
> I don't understand why, because I configured to get a ban after three fails and I get banned after two fails.

Filter sshd is very complex, has many rules comprised different connection/authorization scenarios (related to sshd version, auth-methods, distribution etc).
So it looks like your attempt produced 3 messages in auth.log (which will be recognized as failures in sense of sshd-filter) per single (or two) attempts.
Additionally this is mostly changing filter of fail2ban (due to several reasons) and different from version to version.

Therefore, firstly (I'm tired to repeat it) - for God's sake why you are ignoring out issue template?
- fail2ban version? sshd (system) version?
- log excerpt (related log messages from `/var/log/auth.log` corresponding your attempts)?
- output of `fail2ban-regex -vv` over this excerpt or `/var/log/auth.log`?

> Why does the "currently failed" not increase? And only the total failed increases?

Value of "Currently failed" corresponds to the active tickets that are gathered (IPs with found failures < `maxretry` during `now - findtime` interval), but not yet banned.
If the ticket gets banned it is moved from fail-list to ban-list, so belongs to the "Currently banned".

**Comment by pisoto** [comment]
> > I don't understand why, because I configured to get a ban after three fails and I get banned after two fails.
> 
> Filter sshd is very complex, has many rules comprised different connection/authorization scenarios (related to sshd version, auth-methods, distribution etc).
> So it looks like your attempt produced 3 messages in auth.log (which will be recognized as failures in sense of sshd-filter) per single (or two) attempts.
> Additionally this is mostly changing filter of fail2ban (due to several reasons) and different from version to version.
> 
> Therefore, firstly (I'm tired to repeat it) - for God's sake why you are ignoring out issue template?
> 
> * fail2ban version? sshd (system) version?
> * log excerpt (related log messages from `/var/log/auth.log` corresponding your attempts)?
> * output of `fail2ban-regex -vv` over this excerpt or `/var/log/auth.log`?
> 
> > Why does the "currently failed" not increase? And only the total failed increases?
> 
> Value of "Currently failed" corresponds to the active tickets that are gathered (IPs with found failures < `maxretry` during `now - findtime` interval), but not yet banned.
> If the ticket gets banned it is moved from fail-list to ban-list, so belongs to the "Currently banned".

fail2ban version:
0.11.0.3
sshd version:
OpenSSH_7.6p1 Ubuntu-4ubuntu0.3, OpenSSL 1.0.2n  7 Dec 2017
fail2ban-regex -VV:
0.11.0.3
auth.log:
Aug 20 10:13:01 pisoto-Latitude-E6430 sshd[6937]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.43.250  user=pisoto
Aug 20 10:13:03 pisoto-Latitude-E6430 sshd[6937]: Failed password for pisoto from 192.168.43.250 port 44904 ssh2

**Comment by sebres** [comment]
> fail2ban-regex -VV:
> 0.11.0.3

Nope... I meant:
```
fail2ban-regex -vv /var/log/auth.log sshd
```

> auth.log

I see only two message. I'm pretty sure you'd have more as only this 2 messages (e. g. try to grep by  `sshd[6937]`).

But, OK.
This 2 messages will be recognized by your filter as 2 failures (not as single attempt)... 

Fail2ban does not counts sessions, because it can be more as one authentication attempt per session.

But exactly this one should be fixed in current 0.10 / 0.11 branch (I think in #2090).
So if you take a look into new filter
https://github.com/fail2ban/fail2ban/blob/4ee07adde687e619f459b2e34d795f6efd9ca2f7/config/filter.d/sshd.conf#L52
you will see the rule for first line is marked with tag `<F-NOFAIL>`, which is simply a helper to catch IP for failure messages without IP, and it causes a failure only if no other messages are recognized as failure or in worse cases like user name is switched (so some legitimate user really tried to authenticate with different user name).

Here is PoC:
<details>

```
fail2ban-regex -vv /tmp/gh-2506.txt sshd

Running tests
=============

Use   failregex filter file : sshd, basedir: /etc/fail2ban/
Real  filter options : {'logtype': 'file', 'datepattern': '{^LN-BEG}'}
Use         maxlines : 1
Use      datepattern : Default Detectors
Use         log file : /tmp/gh-2506.txt
Use         encoding : UTF-8


Results
=======

Failregex: 2 total
|-  #) [# of hits] regular expression
|...
|   4) [1] ^Failed \b(?!publickey)\S+ for (?P<cond_inv>invalid user )?<F-USER>(?P<cond_user>\S+)|(?(cond_inv)(?:(?! from ).)*?|[^:]+)</F-USER> from <HOST>(?: (?:port \d+|on \S+)){0,2}(?: ssh\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)
|      192.168.43.250  Tue Aug 20 10:13:03 2019
|...
|  14) [1] ^<F-NOFAIL>pam_[a-z]+\(sshd:auth\):\s+authentication failure;</F-NOFAIL>(?:\s+(?:(?:logname|e?uid|tty)=\S*)){0,4}\s+ruser=<F-ALT_USER>\S*</F-ALT_USER>\s+rhost=<HOST>(?:\s+user=<F-USER>\S*</F-USER>)?(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|      192.168.43.250  Tue Aug 20 10:13:01 2019
|...
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [2] {^LN-BEG}(?:DAY )?MON Day %k:Minute:Second(?:\.Microseconds)?(?: ExYear)?
|...
`-

Lines: 2 lines, 1 ignored, 1 matched, 0 missed
[processed in 0.00 sec]

|- Ignored line(s):
|  Aug 20 10:13:01 pisoto-Latitude-E6430 sshd[6937]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.43.250  user=pisoto
`-
```
</details>

As you can see it says that first message is ignored (so fail2ban does not catch it as a failure).

So you have a choice:
- either to update your fail2ban
- or to increase a `maxretry`
- or to disable pam-related rules, e. g. remove or comment it in filter with `#` or use your own filter.

Because you use v.0.11 you can also enable `bantime.increment` and safe increase `maxretry` and `findtime`, because attempt counter will be automatically incremented for bad (recidive) IPs, which trying it again and again, so such IPs will reach `maxretry` earlier.

**Comment by pisoto** [comment]
> ```
> fail2ban-regex -vv /var/log/auth.log sshd
> ```
> 
> > 
Running tests


Use   failregex filter file : sshd, basedir: /etc/fail2ban
Real  filter options : {'logtype': 'file', 'datepattern': '{^LN-BEG}'}
Use         maxlines : 1
Use      datepattern : Default Detectors
Use         log file : /var/log/auth.log
Use         encoding : UTF-8


Results


Failregex: 47 total
|-  #) [# of hits] regular expression
|   1) [0] ^[aA]uthentication (?:failure|error|failed) for <F-USER>.*</F-USER> from <HOST>( via \S+)?(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|   2) [0] ^User not known to the underlying authentication module for <F-USER>.*</F-USER> from <HOST>(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|   3) [0] ^Failed publickey for invalid user <F-USER>(?P<cond_user>\S+)|(?:(?! from ).)*?</F-USER> from <HOST>(?: (?:port \d+|on \S+)){0,2}(?: ssh\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)
|   4) [23] ^Failed \b(?!publickey)\S+ for (?P<cond_inv>invalid user )?<F-USER>(?P<cond_user>\S+)|(?(cond_inv)(?:(?! from ).)*?|[^:]+)</F-USER> from <HOST>(?: (?:port \d+|on \S+)){0,2}(?: ssh\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)
|      192.168.43.250  Mon Aug 19 07:58:31 2019
|      192.168.43.250  Mon Aug 19 08:16:23 2019
|      192.168.43.250  Mon Aug 19 08:51:19 2019
|      192.168.43.250  Mon Aug 19 09:01:09 2019
|      192.168.43.250  Mon Aug 19 09:02:49 2019
|      192.168.43.250  Mon Aug 19 09:07:17 2019
|      192.168.43.250  Mon Aug 19 09:56:13 2019
|      192.168.43.250  Mon Aug 19 09:57:59 2019
|      192.168.43.250  Mon Aug 19 10:41:58 2019
|      192.168.43.250  Mon Aug 19 10:43:14 2019
|      192.168.43.250  Mon Aug 19 10:43:30 2019
|      192.168.43.250  Mon Aug 19 10:44:33 2019
|      192.168.43.250  Tue Aug 20 04:17:17 2019
|      192.168.43.250  Tue Aug 20 04:17:21 2019
|      192.168.43.250  Tue Aug 20 04:17:53 2019
|      192.168.43.250  Tue Aug 20 04:37:43 2019
|      192.168.43.250  Tue Aug 20 04:38:04 2019
|      192.168.43.250  Tue Aug 20 04:42:01 2019
|      192.168.43.250  Tue Aug 20 04:43:03 2019
|      192.168.43.250  Tue Aug 20 08:38:53 2019
|      192.168.43.250  Tue Aug 20 09:57:04 2019
|      192.168.43.250  Tue Aug 20 10:13:03 2019
|      192.168.43.250  Tue Aug 20 10:14:13 2019
|   5) [0] ^<F-USER>ROOT</F-USER> LOGIN REFUSED FROM <HOST>
|   6) [1] ^[iI](?:llegal|nvalid) user <F-USER>.*?</F-USER> from <HOST>(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|      1.2.3.4  Mon Aug 19 07:35:15 2019
|   7) [0] ^User <F-USER>.+</F-USER> from <HOST> not allowed because not listed in AllowUsers(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|   8) [0] ^User <F-USER>.+</F-USER> from <HOST> not allowed because listed in DenyUsers(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|   9) [0] ^User <F-USER>.+</F-USER> from <HOST> not allowed because not in any group(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|  10) [0] ^refused connect from \S+ \(<HOST>\)
|  11) [0] ^Received <F-MLFFORGET>disconnect</F-MLFFORGET> from <HOST>(?: (?:port \d+|on \S+)){0,2}:\s*3: .*: Auth fail(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|  12) [0] ^User <F-USER>.+</F-USER> from <HOST> not allowed because a group is listed in DenyGroups(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|  13) [0] ^User <F-USER>.+</F-USER> from <HOST> not allowed because none of user's groups are listed in AllowGroups(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|  14) [15] ^<F-NOFAIL>pam_[a-z]+\(sshd:auth\):\s+authentication failure;</F-NOFAIL>(?:\s+(?:(?:logname|e?uid|tty)=\S*)){0,4}\s+ruser=<F-ALT_USER>\S*</F-ALT_USER>\s+rhost=<HOST>(?:\s+user=<F-USER>\S*</F-USER>)?(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|      192.168.43.250  Mon Aug 19 07:58:29 2019
|      192.168.43.250  Mon Aug 19 08:16:21 2019
|      192.168.43.250  Mon Aug 19 08:51:17 2019
|      192.168.43.250  Mon Aug 19 09:01:07 2019
|      192.168.43.250  Mon Aug 19 09:07:15 2019
|      192.168.43.250  Mon Aug 19 09:56:10 2019
|      192.168.43.250  Mon Aug 19 09:57:56 2019
|      192.168.43.250  Mon Aug 19 10:41:57 2019
|      192.168.43.250  Mon Aug 19 10:43:28 2019
|      192.168.43.250  Tue Aug 20 04:17:15 2019
|      192.168.43.250  Tue Aug 20 04:37:41 2019
|      192.168.43.250  Tue Aug 20 04:41:59 2019
|      192.168.43.250  Tue Aug 20 08:38:52 2019
|      192.168.43.250  Tue Aug 20 09:57:02 2019
|      192.168.43.250  Tue Aug 20 10:13:01 2019
|  15) [0] ^(error: )?maximum authentication attempts exceeded for <F-USER>.*</F-USER> from <HOST>(?: (?:port \d+|on \S+)){0,2}(?: ssh\d*)?(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|  16) [0] ^User <F-USER>.+</F-USER> not allowed because account is locked(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*
|  17) [0] ^<F-MLFFORGET>Disconnecting</F-MLFFORGET>(?: from)?(?: (?:invalid|authenticating)) user <F-USER>\S+</F-USER> <HOST>(?: (?:port \d+|on \S+)){0,2}:\s*Change of username or service not allowed:\s*.*\[preauth\]\s*$
|  18) [0] ^<F-MLFFORGET>Disconnecting</F-MLFFORGET>: Too many authentication failures(?: for <F-USER>.+?</F-USER>)?(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|  19) [1] ^<F-NOFAIL>Received <F-MLFFORGET>disconnect</F-MLFFORGET></F-NOFAIL> from <HOST>(?: (?:port \d+|on \S+)){0,2}:\s*11:
|      192.168.43.250  Mon Aug 19 09:04:24 2019
|  20) [6] ^<F-NOFAIL>Connection <F-MLFFORGET>closed</F-MLFFORGET></F-NOFAIL> by(?: authenticating user <F-USER>\S+|.+?</F-USER>)? <HOST>(?:(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*|\s*)$
|      192.168.43.250  Mon Aug 19 08:16:35 2019
|      192.168.43.250  Mon Aug 19 09:57:25 2019
|      192.168.43.250  Mon Aug 19 10:43:14 2019
|      192.168.43.250  Mon Aug 19 10:44:33 2019
|      192.168.43.250  Tue Aug 20 04:17:53 2019
|      192.168.43.250  Tue Aug 20 04:43:03 2019
|  21) [1] ^<F-MLFFORGET><F-MLFGAINED>Accepted \w+</F-MLFGAINED></F-MLFFORGET> for <F-USER>\S+</F-USER> from <HOST>(?:\s|$)
|      192.168.43.250  Mon Aug 19 09:04:17 2019
|  22) [0] ^<F-NOFAIL>Connection from</F-NOFAIL> <HOST>
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [1984] {^LN-BEG}(?:DAY )?MON Day %k:Minute:Second(?:\.Microseconds)?(?: ExYear)?
|  [0] {^LN-BEG}ExYear(?P<_sep>[-/.])Month(?P=_sep)Day(?:T|  ?)24hour:Minute:Second(?:[.,]Microseconds)?(?:\s*Zone offset)?
|  [0] {^LN-BEG}(?:DAY )?MON Day ExYear %k:Minute:Second(?:\.Microseconds)?
|  [0] {^LN-BEG}Day(?P<_sep>[-/])Month(?P=_sep)(?:ExYear|ExYear2) %k:Minute:Second
|  [0] {^LN-BEG}Day(?P<_sep>[-/])MON(?P=_sep)ExYear[ :]?24hour:Minute:Second(?:\.Microseconds)?(?: Zone offset)?
|  [0] {^LN-BEG}Month/Day/ExYear:24hour:Minute:Second
|  [0] {^LN-BEG}Month-Day-ExYear %k:Minute:Second(?:\.Microseconds)?
|  [0] {^LN-BEG}Epoch
|  [0] {^LN-BEG}ExYear2ExMonthExDay  ?24hour:Minute:Second
|  [0] {^LN-BEG}MON Day, ExYear 12hour:Minute:Second AMPM
|  [0] {^LN-BEG}ExYearExMonthExDay(?:T|  ?)Ex24hourExMinuteExSecond(?:[.,]Microseconds)?(?:\s*Zone offset)?
|  [0] {^LN-BEG}(?:Zone name )?(?:DAY )?MON Day %k:Minute:Second(?:\.Microseconds)?(?: ExYear)?
|  [0] {^LN-BEG}(?:Zone offset )?(?:DAY )?MON Day %k:Minute:Second(?:\.Microseconds)?(?: ExYear)?
|  [0] {^LN-BEG}TAI64N
|  [0] {^LN-BEG}24hour:Minute:Second
|  [0] ^<Month/Day/ExYear2@24hour:Minute:Second>
|  [0] ^MON-Day-ExYear2 %k:Minute:Second
`-

Lines: 1985 lines, 23 ignored, 24 matched, 1938 missed
[processed in 0.26 sec]

Ignored line(s): too many to print.  Use --print-all-ignored to print all 23 lines
Missed line(s): too many to print.  Use --print-all-missed to print all 1938 lines
 
 

> But exactly this one should be fixed in current 0.10 / 0.11 branch (I think in #2090) 
> you will see the rule for first line is marked with tag `<F-NOFAIL>`, which is simply a helper to catch IP for failure messages without IP, and it causes a failure only if no other messages are recognized as failure or in worse cases like user name is switched (so some legitimate user really tried to authenticate with different user name).

 ^<F-NOFAIL>%(__pam_auth)s\(sshd:auth\):\s+authentication failure;</F-NOFAIL>(?:\s+(?:(?:logname|e?uid|tty)=\S*)){0,4}\s+ruser=<F-ALT_USER>\S*</F-ALT_USER>\s+rhost=<HOST>(?:\s+user=<F-USER>\S*</F-USER>)?%(__suff)s$

I already use it because I installed the fail2ban from here
and I have the same filter that I cloned with git 
 
> * either to update your fail2ban
>> I have the last version you have here: 0.11.0.3
> * or to increase a `maxretry`
>> but if I want user will get ban only after three wrong passwords 
> * or to disable pam-related rules, e. g. remove or comment it in filter with `#` or use your own filter.
>> I cant remove pam rule because I have password rule that I must to have

**Comment by sebres** [comment]
> I already use it because I installed the fail2ban from here and I have the same filter that I cloned with git

Oops... unexpected. Will check that later. Thx.

**Comment by sebres** [comment]
Can you provide a bit more detailed excerpt from auth.log?
I mean from start of your attempts up to the ban...
(please **don't grep by IP**, because some messages does not contain it, but can be captured as a failure, because filter sshd is a multi-line filter).

**Comment by pisoto** [comment]
> Can you provide a bit more detailed excerpt from auth.log?
> I mean from start of your attempts up to the ban...
> (please **don't grep by IP**, because some messages does not contain it, but can be captured as a failure, because filter sshd is a multi-line filter).

Aug 22 04:35:55 pisoto-Latitude-E6430 sshd[13079]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.43.250  user=pisoto
Aug 22 04:35:57 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
Aug 22 04:36:02 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
Aug 22 04:36:14 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
Aug 22 04:36:14 pisoto-Latitude-E6430 sshd[13079]: Connection closed by authenticating user pisoto 192.168.43.250 port 44912 [preauth]
Aug 22 04:36:14 pisoto-Latitude-E6430 sshd[13079]: PAM 2 more authentication failures; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.43.250  user=pisoto

**Comment by sebres** [comment]
Thx!
Do I correct understand, you have made exactly 3 attempts here (in this session)?

So by `maxretry = 3` fail2ban has every right to ban it... Correct?...

But, what you meant previously in the issue, it should either look more like:
```
Aug 22 04:35:55 pisoto-Latitude-E6430 sshd[13079]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.43.250  user=pisoto
Aug 22 04:35:57 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
Aug 22 04:36:02 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
***Ban***
```
Because you told about the ban happening after 2 attempts... Am I wrong?

Can you please explain that, or better do your tries together with running this:
```
?sudo? tail -f /var/log/auth.log /var/log/fail2ban.log
```

**Comment by pisoto** [comment]
> Thx!
> Do I correct understand, you have made exactly 3 attempts here (in this session)?
> 
> So by `maxretry = 3` fail2ban has every right to ban it... Correct?...
> 
> But, what you meant previously in the issue, it should either look more like:
> 
> ```
> Aug 22 04:35:55 pisoto-Latitude-E6430 sshd[13079]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.43.250  user=pisoto
> Aug 22 04:35:57 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
> Aug 22 04:36:02 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
> ***Ban***
> ```
> 
> Because you told about the ban happening after 2 attempts... Am I wrong?
> 
> Can you please explain that, or better do your tries together with running this:
> 
> ```
> ?sudo? tail -f /var/log/auth.log /var/log/fail2ban.log
> ```

After I deleted fail2ban and installed it(on first run it is looko good). i am trying to put 3 wrong password, and it is working fine, and sometimes the counter starts from 2. but after the first ban end when I am trying to get ban again I need to get ban after 3 times i am getting ban after one time, and in the fail2ban-client status sshd it is writing 2 fails and give ban for only one wrong password, it adds more than one every time to the counter

**Comment by pisoto** [comment]
> Thx!
> Do I correct understand, you have made exactly 3 attempts here (in this session)?
> 
> So by `maxretry = 3` fail2ban has every right to ban it... Correct?...
> 
> But, what you meant previously in the issue, it should either look more like:
> 
> ```
> Aug 22 04:35:55 pisoto-Latitude-E6430 sshd[13079]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.43.250  user=pisoto
> Aug 22 04:35:57 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
> Aug 22 04:36:02 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
> ***Ban***
> ```
> 
> Because you told about the ban happening after 2 attempts... Am I wrong?
> 
> Can you please explain that, or better do your tries together with running this:
> 
> ```
> ?sudo? tail -f /var/log/auth.log /var/log/fail2ban.log
> ```

After I deleted fail2ban and installed it(on first run it looks good). I am trying to put 3 wrong passwords, and it is working fine, and sometimes the counter starts from 2. but after the first ban end when I am trying to get ban again I need to get ban after 3 times I am getting ban after one time, and in the fail2ban-client status sshd it is writing 2 fails and give ban for only one wrong password, it adds more than one every time to the counter

**Comment by pisoto** [comment]
> Thx!
> Do I correct understand, you have made exactly 3 attempts here (in this session)?
> 
> So by `maxretry = 3` fail2ban has every right to ban it... Correct?...
> 
> But, what you meant previously in the issue, it should either look more like:
> 
> ```
> Aug 22 04:35:55 pisoto-Latitude-E6430 sshd[13079]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.43.250  user=pisoto
> Aug 22 04:35:57 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
> Aug 22 04:36:02 pisoto-Latitude-E6430 sshd[13079]: Failed password for pisoto from 192.168.43.250 port 44912 ssh2
> ***Ban***
> ```
> 
> Because you told about the ban happening after 2 attempts... Am I wrong?
> 
> Can you please explain that, or better do your tries together with running this:
> 
> ```
> ?sudo? tail -f /var/log/auth.log /var/log/fail2ban.log
> ```

here is the best example I reseted the counters of fails and bans(sudo systemctl restart fail2ban) set in the config file maxretry of 5 and i get ban after 2 wrong passwords
this is the log:
==> /var/log/auth.log <==
Aug 22 14:12:04 pisoto-Latitude-E6430 sudo: pam_unix(sudo:session): session opened for user root by (uid=0)
Aug 22 14:13:31 pisoto-Latitude-E6430 sshd[10868]: Connection closed by authenticating user pisoto 192.168.43.250 port 48430 [preauth]
Aug 22 14:13:37 pisoto-Latitude-E6430 sudo: pam_unix(sudo:session): session closed for user root
Aug 22 14:13:42 pisoto-Latitude-E6430 sudo:   pisoto : TTY=pts/2 ; PWD=/home/pisoto ; USER=root ; COMMAND=/usr/bin/tail -f /var/log/auth.log /var/log/fail2ban.log
Aug 22 14:13:42 pisoto-Latitude-E6430 sudo: pam_unix(sudo:session): session opened for user root by (uid=0)
Aug 22 14:13:46 pisoto-Latitude-E6430 sudo: pam_unix(sudo:session): session closed for user root
Aug 22 14:14:15 pisoto-Latitude-E6430 sudo:   pisoto : TTY=pts/2 ; PWD=/home/pisoto ; USER=root ; COMMAND=/usr/bin/tail -f /var/log/auth.log /var/log/fail2ban.log
Aug 22 14:14:15 pisoto-Latitude-E6430 sudo: pam_unix(sudo:session): session opened for user root by (uid=0)
Aug 22 14:14:20 pisoto-Latitude-E6430 sudo: pam_unix(sudo:session): session closed for user root
Aug 22 14:14:31 pisoto-Latitude-E6430 sudo:   pisoto : TTY=pts/2 ; PWD=/home/pisoto ; USER=root ; COMMAND=/usr/bin/tail -f /var/log/auth.log /var/log/fail2ban.log

==> /var/log/fail2ban.log <==
2019-08-22 14:11:24,256 fail2ban.jail           [10823]: INFO    Creating new jail 'sshd'
2019-08-22 14:11:24,347 fail2ban.jail           [10823]: INFO    Jail 'sshd' uses poller {}
2019-08-22 14:11:24,347 fail2ban.jail           [10823]: INFO    Initiated 'polling' backend
2019-08-22 14:11:24,372 fail2ban.filter         [10823]: INFO      maxLines: 1
2019-08-22 14:11:24,399 fail2ban.filter         [10823]: INFO    Added logfile: '/var/log/auth.log' (pos = 286192, hash = 31262be32e4034da6f1887e2fd62b5d1)
2019-08-22 14:11:24,399 fail2ban.filter         [10823]: INFO      maxRetry: 5
2019-08-22 14:11:24,399 fail2ban.filter         [10823]: INFO      encoding: UTF-8
2019-08-22 14:11:24,400 fail2ban.actions        [10823]: INFO      banTime: 60
2019-08-22 14:11:24,400 fail2ban.filter         [10823]: INFO      findtime: 600
2019-08-22 14:11:24,401 fail2ban.jail           [10823]: INFO    Jail 'sshd' started

==> /var/log/auth.log <==
Aug 22 14:14:31 pisoto-Latitude-E6430 sudo: pam_unix(sudo:session): session opened for user root by (uid=0)
Aug 22 14:14:45 pisoto-Latitude-E6430 sshd[10898]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.43.250  user=pisoto
Aug 22 14:14:47 pisoto-Latitude-E6430 sshd[10898]: Failed password for pisoto from 192.168.43.250 port 48432 ssh2

==> /var/log/fail2ban.log <==
2019-08-22 14:14:47,522 fail2ban.filter         [10823]: INFO    [sshd] Found 192.168.43.250 - 2019-08-22 14:14:47
2019-08-22 14:14:47,526 fail2ban.observer       [10823]: INFO    [sshd] Found 192.168.43.250, bad - 2019-08-22 14:14:47, 2 # -> 3

==> /var/log/auth.log <==
Aug 22 14:14:53 pisoto-Latitude-E6430 sshd[10898]: Failed password for pisoto from 192.168.43.250 port 48432 ssh2

==> /var/log/fail2ban.log <==
2019-08-22 14:14:53,531 fail2ban.filter         [10823]: INFO    [sshd] Found 192.168.43.250 - 2019-08-22 14:14:53
2019-08-22 14:14:53,553 fail2ban.observer       [10823]: INFO    [sshd] Found 192.168.43.250, bad - 2019-08-22 14:14:53, 2 # -> 3
2019-08-22 14:14:53,857 fail2ban.actions        [10823]: NOTICE  [sshd] Ban 192.168.43.250
2019-08-22 14:15:53,974 fail2ban.actions        [10823]: NOTICE  [sshd] Unban 192.168.43.250

**Comment by pisoto** [comment]
Why did you close this issue?

**Comment by sebres** [comment]
> I am trying to put 3 wrong passwords, and it is working fine, and sometimes the counter starts from 2.

Oh... newer used 0.11 without `bantime.increment`...
Thus fixed in 6b7825b8c80ffe24088f2cbce4ea75aaa60ceded. Thx!

The fail2ban had mistakenly used "incremental" banning, so IPs known as "bad" (banned more as one time) will become next ban faster (and ban will be longer).

See https://github.com/fail2ban/fail2ban/pull/716#issuecomment-43067425

>  I reseted the counters of fails and bans(sudo systemctl restart fail2ban) 

This changes nothing because "bad" IPs (with counter per IP) are stored in database of fail2ban.
So ban-count reset (unmark bad IP) is possible only using manual unban (via fail2ban-client) or manipulating (or deleting fail2ban database) or after auto-purge process.

**Comment by sebres** [comment]
> Why did you close this issue?

This is closed automatically by github, if commit referencing issue gets merged (pushed) into repository.
The issue should be fixed now.

**Comment by pisoto** [comment]
> referencing

oh I understand sorry

**Comment by pisoto** [comment]
> > Why did you close this issue?
> 
> This is closed automatically by github, if commit referencing issue gets merged (pushed) into repository.
> The issue should be fixed now.

oh i understand sorry

**Comment by pisoto** [comment]
> > I am trying to put 3 wrong passwords, and it is working fine, and sometimes the counter starts from 2.
> 
> Oh... newer used 0.11 without `bantime.increment`...
> Thus fixed in [6b7825b](https://github.com/fail2ban/fail2ban/commit/6b7825b8c80ffe24088f2cbce4ea75aaa60ceded). Thx!
> 
> The fail2ban had mistakenly used "incremental" banning, so IPs known as "bad" (banned more as one time) will become next ban faster (and ban will be longer).
> 
> See [#716 (comment)](https://github.com/fail2ban/fail2ban/pull/716#issuecomment-43067425)
> 
> > I reseted the counters of fails and bans(sudo systemctl restart fail2ban)
> 
> This changes nothing because "bad" IPs (with counter per IP) are stored in database of fail2ban.
> So ban-count reset (unmark bad IP) is possible only using manual unban (via fail2ban-client) or manipulating (or deleting) fail2ban database or after regular auto-purge process.

But it is not "bantime" problem it is the number of fails to ban.
After the IP get ban once, the user can wrong the password only one more time every time.
Where can I find the config of maxretry after the first ban?

**Comment by sebres** [comment]
> But it is not "bantime" problem it is the number of fails to ban.

Yes, just with incremental banning it affects both. It is bancount related, so the "bad" IP is the longer the ban and the faster it will be banned (needs fewer attempts).
So by enabled increment (but the last was mistakenly used also by disabled, which was the issue and fixed now). 

> After the IP get ban once, the user can wrong the password only one more time every time.

Yes, but now only if incremental banning enabled. Still again - it is fixed.

> Where can I find the config of maxretry after the first ban?

Read my comment link above (https://github.com/fail2ban/fail2ban/pull/716#issuecomment-43067425).
This is not configurable at the moment (will always do it this way if `bantime.increment` is enabled).
Therefore it is advisable to increase `maxretry ` for incremental banning (bad IPs going anywhere faster banned).
